From a98c96c1ff6e8e6486ce85b7d7becf70bba07b81 Mon Sep 17 00:00:00 2001
From: David Haines <haines@berkshiregrey.com>
Date: Sun, 5 Jan 2020 23:13:45 -0500
Subject: [PATCH] Allow for equals characters in config URLs

---
 args/args.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/args/args.go b/args/args.go
index afc9a92..d2edddc 100644
--- a/args/args.go
+++ b/args/args.go
@@ -107,7 +107,7 @@ func (args *Args) setKernelArgs() (err error) {
 	for _, curr := range strings.Split(kernelCmd, " ") {
 		curr = strings.TrimSpace(curr)
 		if strings.HasPrefix(curr, kernelCmdlineConf+"=") {
-			url = strings.Split(curr, "=")[1]
+			url = strings.Join(strings.Split(curr, "=")[1:], "=")
 		} else if strings.HasPrefix(curr, kernelCmdlineDemo) {
 			args.DemoMode = true
 		} else if strings.HasPrefix(curr, kernelCmdlineLog) {
-- 
2.24.1

